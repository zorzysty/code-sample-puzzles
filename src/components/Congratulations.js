import * as PropTypes from 'prop-types';
import React from 'react';


const Congratulations = ({ visible }) => (
  visible && <div className="Congratulations">Awesome! You won!!!</div>
);

Congratulations.propTypes = { visible: PropTypes.bool.isRequired };

export default Congratulations;
