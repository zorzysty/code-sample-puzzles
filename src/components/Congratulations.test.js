import React from 'react';
import renderer from 'react-test-renderer';
import Congratulations from './Congratulations';


describe('Congratulations component', () => {
  it('should render correctly based on passed props', () => {
    const propsArray = [
      {
        visible: false,
      },
      {
        visible: true,
      },
    ];

    propsArray.forEach((props) => {
      const component = renderer.create(<Congratulations {...props} />);
      expect(component.toJSON())
        .toMatchSnapshot();
    });
  });
});
