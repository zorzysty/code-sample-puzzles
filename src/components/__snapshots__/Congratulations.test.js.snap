// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Congratulations component should render correctly based on passed props 1`] = `null`;

exports[`Congratulations component should render correctly based on passed props 2`] = `
<div
  className="Congratulations"
>
  Awesome! You won!!!
</div>
`;
